### PR TITLE
Buffs His Grace

### LINKED ISSUE #19605: His Grace continues to eat its user upon satiation.
**Issue Description**:
His Grace continues to tally blood lust for each person to kill once sated.

**What did you expect to happen**: 
Blood lust to pause upon somehow killing 25 people.

**What happened instead**:
I got eated.

**Why is this bad/What are the consequences**:
I should be able to kill 150 people without my green god trying to murder me.

**Steps to reproduce the problem**:
Spawn in his grace. Kill 25 skrells. Kill 150 more.

**When did the problem start happening**:
Probably when New His Grace was merged.

**Extra information**:
:(

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do

- Lowers the amount of bloodthirst you get every second. You get about 15 to 20 seconds for the last 5 bodies, with this change, you get 36 to 31 seconds instead, with the earlier numbers being higher aswell of course.
- Adds a counter on the status effect hover (That many probably didn't know existed) of how far you are with your ascension.
- Ascension now sets your bloodthirst to 140, greatly buffing the amount you regenerate.
- Buffs the ascended bonus damage by 1, going from 19 per corpse to 20

## Why It's Good For The Game
His Grace is a very decent item, meant for hijacking chaplains. However, as it currently is, it's very hard to get the required bodies to ascend. You get about 15 to 20 seconds per body in the final 5 victims required to ascend, which is very tough as everyone is probably shoving you away, or is trying to bash your skull in.
The ascending buffs are there so you actually have a good reason to. Bloodthirst being at 140 means that you get 7 of each damage type healing, and 14 oxyloss, healing per second, as the amount scales with how much your bloodthirst is.

Resolves #19605 by making it a lot easier and more possible to actually ascend. Also a counter to check if you should or should not have ascended already.

## Testing
I beat up a lot of skrell, saw His Grace ascend, dropped it and then let it eat me. It didn't eat me after ascension.

## Changelog
:cl:
tweak: His Grace will now get less Bloodthirst per second.
tweak: His Grace now gets even stronger on ascension, and the holder gets more regeneration.
fix: Fixed some various His Grace bugs.
/:cl:

### CODE REVIEW

File: code/__DEFINES/combat_defines.dm [hunk 1]
```diff
@@ -129,17 +129,6 @@
...
-//His Grace.
-#define HIS_GRACE_SATIATED 0 //He hungers not. If bloodthirst is set to this, His Grace is asleep.
-#define HIS_GRACE_PECKISH 20 //Slightly hungry.
-#define HIS_GRACE_HUNGRY 60 //Getting closer. Increases damage up to a minimum of 20.
-#define HIS_GRACE_FAMISHED 100 //Dangerous. Increases damage up to a minimum of 25 and cannot be dropped.
-#define HIS_GRACE_STARVING 120 //Incredibly close to breaking loose. Increases damage up to a minimum of 30.
-#define HIS_GRACE_CONSUME_OWNER 140 //His Grace consumes His owner at this point and becomes aggressive.
-#define HIS_GRACE_FALL_ASLEEP 160 //If it reaches this point, He falls asleep and resets.
-
-#define HIS_GRACE_FORCE_BONUS 4 //How much force is gained per kill.
```
Comment (Sirryan2002) [inline code/__DEFINES/combat_defines.dm:141]: why move this out of defines if you're converting the defines back into magic numbers in another file such as `buffs.dm` ?

File: code/game/objects/items/his_grace.dm [hunk 2]
```diff
@@ -182,30 +208,39 @@
...
 	meal.adjustBruteLoss(300)
 	playsound(meal, 'sound/misc/desceration-02.ogg', 75, TRUE)
 	playsound(src, 'sound/items/eatfood.ogg', 100, TRUE)
 	meal.forceMove(src)
-	force_bonus += HIS_GRACE_FORCE_BONUS
+
+	if(ascended)
+		force_bonus += ASCEND_BONUS
+	else
+		force_bonus += HIS_GRACE_FORCE_BONUS
```
Comment (Sirryan2002) [inline code/game/objects/items/his_grace.dm:221]:
```suggestion
	force_bonus += ascended ? ASCEND_BONUS : HIS_GRACE_FORCE_BONUS
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
looks pretty good, i spotted this though

File: code/game/objects/items/his_grace.dm [hunk 3]
```diff
@@ -69,23 +72,24 @@
...
-			if(HIS_GRACE_FAMISHED to HIS_GRACE_STARVING)
-				. += "<span class='his_grace'>[src] is openly salivating at the sight of you. Be careful.</span>"
-			if(HIS_GRACE_STARVING to HIS_GRACE_CONSUME_OWNER)
-				. += "<span class='his_grace bold'>You walk a fine line. [src] is very close to devouring you.</span>"
-			if(HIS_GRACE_CONSUME_OWNER to HIS_GRACE_FALL_ASLEEP)
-				. += "<span class='his_grace bold'>[src] is shaking violently and staring directly at you.</span>"
-	else
+
+	if(!awakened)
 		. += "<span class='his_grace'>[src] is latched closed.</span>"
```
Comment (Contrabang) [inline code/game/objects/items/his_grace.dm:77]:
```suggestion
		. += "<span class='his_grace'>[src] is latched closed.</span>"
		return
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
One final thing

**Review by Contrabang — APPROVED** [review]

File: code/datums/status_effects/buffs.dm [hunk 4]
```diff
@@ -33,6 +36,8 @@
 	held_items += owner.l_hand
 	held_items += owner.r_hand
 	for(var/obj/item/his_grace/HG in held_items)
+		victims = HG.victims
+		victims_needed = HG.victims_needed
```
Comment (lewcc) [inline code/datums/status_effects/buffs.dm:40]: I'm not a huge fan of mirroring values from one set of variables to another like this, it feels like there should only be one source of truth. It almost feels to me like it'd be nicer to include a reference to the toolbox in the status effect and just get the values right from it? It shouldn't be too hard to make something like this GC nicely. Regardless, it's probably worth another look.

**Review by Contrabang — CHANGES_REQUESTED** [review]
See Lewc's comment

File: code/game/objects/items/his_grace.dm [hunk 5]
```diff
@@ -95,29 +100,40 @@
...
-				master.remove_status_effect(STATUS_EFFECT_HISGRACE)
-				flags &= ~NODROP
-				master.Weaken(6 SECONDS)
-				master.adjustBruteLoss(1000)
-				playsound(master, 'sound/effects/splat.ogg', 100, FALSE)
-			else
-				master.apply_status_effect(STATUS_EFFECT_HISGRACE)
+	held_items += master.l_hand
+	held_items += master.r_hand
+	if(!(src in held_items))
```
[the remark this answers is not in the document]
Comment (lewcc) [reply code/game/objects/items/his_grace.dm:116]: oh god what happened to that diff

[the remark this answers is not in the document]
Comment (DGamerL) [reply code/game/objects/items/his_grace.dm:116]: I have no idea and I'm scared

**Review by lewcc — APPROVED** [review]

**Review by Contrabang — APPROVED** [review]

**Pull request merged**